packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -833,7 +748,10 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 					dynamicRetries: true, // To differentiate this telemetry from regular retry logic
 					...attemptResult.summarizeProps,
 				});
-				await delay(retryAfterSeconds * 1000);
+				// If "retryAfterSeconds" is 0, no need to delay.
+				if (retryAfterSeconds > 0) {
```

Comment:
nit: Is it valuable to log "SummarizeAttemptDelay" in the case we don't actually delay?

Reply:
Good poiint. Removed it if it's 0.